src/pages/index.js

```diff
@@ -29,60 +30,155 @@ const enableValidation = (config) => {
 
 enableValidation(validationConfig);
 
-//creating cards from the array
-const popupWithImage = new PopupWithImage(".modal_type_show-image");
+//api
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "9e620f98-b2bc-4f4a-81bd-341fb1cf797f",
+  },
+});
 
-const handleImageClick = (card) => {
-  popupWithImage.open(card.link, card.name);
+//user
+api
+  .getUserInfo()
+  .then((userInfo) => {
+    //set user info
+    const user = new UserInfo(
+      ".explorer__name",
+      ".explorer__description",
+      ".explorer__photo"
+    );
+    user.setUserInfo({
+      name: userInfo.name,
+      about: userInfo.about,
+    });
+    user.setAvatar({ avatar: userInfo.avatar });
+    return user;
+  })
+  .then((user) => {
+    //cahnging user details
+    function saveProfileChanges(newUserData) {
+      profileEditPopup.savingDisplay(true);
+      user.setUserInfo(newUserData);
```

Comment:
You need to update `DOM` only in blocks `then` after a successful response from the server so that when there is an error it wouldn’t change

**Please, correct it everywhere**